contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -0,0 +1,460 @@
+      fallbackNativePerUnitLink,
+      donFee,
+      maxSupportedRequestDataVersion
+    );
+  }
+
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getConfig()
+    external
+    view
+    returns (
+      uint32 maxCallbackGasLimit,
+      uint32 feedStalenessSeconds,
+      uint256 gasOverheadAfterCallback,
+      int256 fallbackNativePerUnitLink,
+      uint32 gasOverheadBeforeCallback,
+      address linkPriceFeed,
+      uint16 maxSupportedRequestDataVersion
+    )
+  {
+    return (
+      s_config.maxCallbackGasLimit,
+      s_config.feedStalenessSeconds,
+      s_config.gasOverheadAfterCallback,
+      s_config.fallbackNativePerUnitLink,
+      s_config.gasOverheadBeforeCallback,
+      address(LINK_TO_NATIVE_FEED),
+      s_config.maxSupportedRequestDataVersion
+    );
+  }
+
+  // ================================================================
+  // |                 Cost Calculation Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getDONFee(
+    bytes memory /* requestData */,
+    RequestBilling memory /* billing */
+  ) public view override returns (uint96) {
+    // NOTE: Optionally, compute additional fee here
+    return s_config.donFee;
+  }
+
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getAdminFee(
+    bytes memory /* requestData */,
+    RequestBilling memory /* billing */
+  ) public view override returns (uint96) {
+    // NOTE: Optionally, compute additional fee here
+    return IFunctionsRouter(address(s_router)).getAdminFee();
+  }
+
+  function getFeedData() public view returns (int256) {
+    uint32 feedStalenessSeconds = s_config.feedStalenessSeconds;
+    bool staleFallback = feedStalenessSeconds > 0;
+    (, int256 weiPerUnitLink, , uint256 timestamp, ) = LINK_TO_NATIVE_FEED.latestRoundData();
+    // solhint-disable-next-line not-rely-on-time
+    if (staleFallback && feedStalenessSeconds < block.timestamp - timestamp) {
+      weiPerUnitLink = s_config.fallbackNativePerUnitLink;
+    }
+    return weiPerUnitLink;
+  }
+
+  // ================================================================
+  // |                  Cost Estimation Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function estimateCost(
+    uint64 subscriptionId,
+    bytes calldata data,
+    uint32 callbackGasLimit,
+    uint256 gasPrice
+  ) external view override returns (uint96) {
+    RequestBilling memory billing = RequestBilling(subscriptionId, msg.sender, callbackGasLimit, gasPrice);
+    uint96 donFee = getDONFee(data, billing);
+    uint96 adminFee = getAdminFee(data, billing);
+    return _calculateCostEstimate(callbackGasLimit, gasPrice, donFee, adminFee);
+  }
+
+  /**
+   * @notice Uses current price feed data to estimate a cost
+   */
+  function _calculateCostEstimate(
```

Comment:
What are we gonna do about the problem of an increase in gas price between time of request and time of response?  Should we add an overestimate percentage for gas price or something?  This could be a config variable.